src/commands/gate/evaluate.ts

```diff
@@ -119,22 +126,21 @@ export class GateEvaluateCommand extends Command {
   }
 
   private async evaluateRules(api: APIHelper, evaluateRequest: Payload): Promise<number> {
-    if (this.shouldWait()) {
-      this.context.stdout.write(renderWaiting())
-      await this.delay(this.waitingTime)
-    }
-
     this.context.stdout.write(renderGateEvaluationInput(evaluateRequest))
 
-    return retryRequest(
-      () => api.evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout)),
-      {
-        onRetry: (e, attempt) => {
+    // retry is handled in evaluateRulesWithWait, so we set maxTimeout and minTimeout as 0 for retryRequest
+    return retryRequest((attempt) => this.evaluateRulesWithWait(api, evaluateRequest, attempt), {
+      onRetry: (e, attempt) => {
+        // render retry message if error is not wait
+        if (e.message !== 'wait') {
           this.context.stderr.write(renderEvaluationRetry(attempt, e))
-        },
-        retries: 5,
-      }
-    )
+        }
+      },
+      maxRetryTime: this.maxRetryTimeMs,
```

Comment:
I think we need not to set this, as this seems to be the _maximum_ time that we'll allow all the retries to last (https://github.com/tim-kos/node-retry/blob/master/lib/retry_operation.js#L10)
```suggestion
```

Reply:
That was my idea, but now I'm thinking about it, we should do it on the backend, to make the timeout more flexible as well.